src/rotation.zig

```diff
@@ -357,35 +357,242 @@ pub const RotationModes = struct {
+				},
+				.bend => {
+					out.append(rotateQuad(.{
+						.{0.0, 0.0},
+						.{0.0, maxF},
+						.{maxF, 0.0},
+						.{maxF, maxF},
+					}, pattern, minF, maxF, side));
+				},
+				.intersection => {
+					out.append(rotateQuad(.{
+						.{0.0, 0.0},
+						.{0.0, maxF},
+						.{1.0, 0.0},
+						.{1.0, maxF},
+					}, pattern, minF, maxF, side));
+				},
+				.cross => {
+					out.append(rotateQuad(.{
+						.{0.0, 0.0},
+						.{0.0, 1.0},
+						.{1.0, 0.0},
+						.{1.0, 1.0},
+					}, pattern, minF, maxF, side));
+				},
 			}
 		}
 
+		fn getPattern(data: BranchData, side: Neighbor) ?Pattern {
+			const posX = Neighbor.fromRelPos(side.textureX()).?;
+			const posNegX = Neighbor.fromRelPos(side.textureX()).?.reverse();
+			const posY = Neighbor.fromRelPos(side.textureY()).?;
+			const posNegY = Neighbor.fromRelPos(side.textureY()).?.reverse();
+
+			const connectedPosX = data.isConnected(posX);
+			const connectedNegX = data.isConnected(posNegX);
+			const connectedPosY = data.isConnected(posY);
+			const connectedNegY = data.isConnected(posNegY);
+
+			const count: u6 = @as(u6, @intFromBool(connectedPosX)) + @as(u6, @intFromBool(connectedNegX)) + @as(u6, @intFromBool(connectedPosY)) + @as(u6, @intFromBool(connectedNegY));
+
+			return switch(count) {
+				0 => {
+					if(data.isConnected(side)) {
+						return null;
+					}
+
+					return .dot;
+				},
+				1 => {
+					var dir: Direction = .negXDir;
+					if(connectedNegY) {
+						dir = .negYDir;
+					} else if(connectedPosX) {
+						dir = .posXDir;
+					} else if(connectedPosY) {
+						dir = .posYDir;
+					}
+					return .{.halfLine = .{.dir = dir}};
+				},
+				2 => {
+					if((connectedPosX and connectedNegX) or (connectedPosY and connectedNegY)) {
+						var dir: Direction = .negYDir;
+						if(connectedPosX and connectedNegX) {
+							dir = .posXDir;
+						}
+
+						return .{.line = .{.dir = dir}};
+					}
+
+					var dir: Direction = .negXDir;
+
+					if(connectedNegY) {
+						dir = .negYDir;
+					}
+
+					if(connectedNegY) {
+						dir = .negYDir;
+						if(connectedPosX) {
+							dir = .posXDir;
+						}
+					} else if(connectedPosX) {
+						dir = .posXDir;
+						if(connectedPosY) {
+							dir = .posYDir;
+						}
+					} else if(connectedPosY) {
+						dir = .posYDir;
+						if(connectedNegX) {
+							dir = .negXDir;
+						}
+					}
+
+					return .{.bend = .{.dir = dir}};
+				},
+				3 => {
+					var dir: Direction = undefined;
+					if(!connectedPosY) dir = .negYDir;
+					if(!connectedNegX) dir = .posXDir;
+					if(!connectedNegY) dir = .posYDir;
+					if(!connectedPosX) dir = .negXDir;
+
+					return .{.intersection = .{.dir = dir}};
+				},
+				4 => {
+					return .cross;
+				},
+				else => undefined,
+			};
+		}
+
 		pub fn createBlockModel(zon: ZonElement) u16 {
-			const modelId = zon.as([]const u8, "cubyz:cube");
-			if(branchModels.get(modelId)) |modelIndex| return modelIndex;
+			const radius = zon.get(u32, "radius", 4);
```

Comment:
Should be an `f32`, while we should not use partial pixels in the base game, there is no need to restrict it for addons.
Furthermore even without that `3.5` (→ 7 pixels) would also make sense.